### PR TITLE
Fix some PHP mistakes

### PR DESCRIPTION
**Pull request by luigifab**
Initialize $newValue for 2 sql upgrade.
Add default parameter to idn_to_ascii().
Fix some PHP mistakes (undefined index, @new, sizeof on null).

### CODE REVIEW

File: app/code/core/Mage/Core/Helper/Url.php [hunk 1]
```diff
@@ -180,7 +180,7 @@ public function encodePunycode($url)
         $parsedUrl = parse_url($url);
         if (!$this->_isPunycode($parsedUrl['host'])) {
             if (function_exists('idn_to_ascii')) {
-                $host = idn_to_ascii($parsedUrl['host']);
+                $host = idn_to_ascii($parsedUrl['host'], IDNA_DEFAULT, INTL_IDNA_VARIANT_UTS46);
```
Comment (spinsch) [inline app/code/core/Mage/Core/Helper/Url.php]: the parameters are already set as default. why must this be specified explicitly now?

File: app/code/core/Mage/AdminNotification/Model/Feed.php [hunk 2]
```diff
@@ -169,7 +169,7 @@ public function getFeedData()
         $curl->close();
 
         try {
-            $xml  = new SimpleXMLElement($data);
+            $xml = @new SimpleXMLElement($data);
```
Comment (spinsch) [inline app/code/core/Mage/AdminNotification/Model/Feed.php]: what happens now if an exception is thrown?

File: app/code/core/Mage/Core/Helper/Url.php (hunk 1 above)
Comment (luigifab) [reply app/code/core/Mage/Core/Helper/Url.php]: I have added these default parameters some months ago, at this time, there was a notice or a warning with php 7.2? (I don't remember exactly). I tested again with php 7.2 & 7.4, and I'm agree these default parameters are useless. Sorry.

File: app/code/core/Mage/AdminNotification/Model/Feed.php (hunk 2 above)
Comment (luigifab) [reply app/code/core/Mage/AdminNotification/Model/Feed.php]: Wow! I'm stupid, ignore that.

File: lib/Zend/Service/WindowsAzure/Storage/Batch.php [hunk 3]
```diff
@@ -105,7 +105,7 @@ protected function _clean()
         unset($this->_operations);
         $this->_storageClient->setCurrentBatch(null);
         $this->_storageClient = null;
-        unset($this);
+        //unset($this);
```
Comment (colinmollenhour) [inline lib/Zend/Service/WindowsAzure/Storage/Batch.php:108]: Code should just be deleted rather than commented out.

**Comment by luigifab** [comment]
I will resend a pull request, I don't understand how to edit.

**Pull request closed without merging**